src/Microsoft.Windows.CsWin32/Generator.cs

```diff
@@ -3154,48 +3176,43 @@ private ParameterSyntax CreateParameter(MethodSignature<TypeSyntax> methodSignat
-                    ////}
-                    MethodDeclarationSyntax getOrSetAtMethod = MethodDeclaration(RefType(qualifiedElementType), "GetOrSetAt")
-                        .WithModifiers(methodModifiers)
-                        .AddParameterListParameters(thisParameter.AddModifiers(Token(SyntaxKind.RefKeyword)), indexParameter)
-                        .WithBody(body);
-                    this.inlineArrayIndexerExtensionsMembers.Add(getOrSetAtMethod);
-                }
+                IdentifierNameSyntax indexParamName = IdentifierName("index");
+                IdentifierNameSyntax p0 = IdentifierName("p0");
+                IdentifierNameSyntax atThis = IdentifierName("@this");
+                TypeSyntax qualifiedElementType = elementType == IntPtrTypeSyntax ? elementType : ((ArrayTypeSyntax)fieldDef.DecodeSignature(this.signatureTypeProviderFullyQualified, null)).ElementType;
+
+                ////internal static unsafe ref readonly uint GetAt(this in MainAVIHeader.__dwReserved_4 @this, int index)
+                ////{
+                ////    fixed (uint* p0 = &@this._1)
+                ////        return ref p0[index];
+                ////}
+                FixedStatementSyntax? fixedStatement = FixedStatement(
+                    VariableDeclaration(PointerType(qualifiedElementType)).AddVariables(
+                        VariableDeclarator(p0.Identifier).WithInitializer(EqualsValueClause(
+                            PrefixUnaryExpression(
+                                SyntaxKind.AddressOfExpression,
+                                MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, atThis, firstElementFieldName))))),
+                    ReturnStatement(RefExpression(ElementAccessExpression(p0).AddArgumentListArguments(Argument(indexParamName)))));
+                BlockSyntax body = Block().AddStatements(fixedStatement);
+                ParameterSyntax thisParameter = Parameter(atThis.Identifier).WithType(QualifiedName(this.GetQualifiedName(fieldDef.GetDeclaringType()), fixedLengthStructName)).AddModifiers(Token(SyntaxKind.ThisKeyword));
+                ParameterSyntax indexParameter = Parameter(indexParamName.Identifier).WithType(PredefinedType(Token(SyntaxKind.IntKeyword)));
+                SyntaxTokenList methodModifiers = TokenList(Token(this.Visibility), Token(SyntaxKind.StaticKeyword), Token(SyntaxKind.UnsafeKeyword));
+                MethodDeclarationSyntax getAtMethod = MethodDeclaration(RefType(qualifiedElementType).WithReadOnlyKeyword(Token(SyntaxKind.ReadOnlyKeyword)), "GetAt")
+                    .WithModifiers(methodModifiers)
+                    .AddParameterListParameters(thisParameter.AddModifiers(Token(SyntaxKind.InKeyword)), indexParameter)
+                    .WithBody(body);
+                this.inlineArrayIndexerExtensionsMembers.Add(getAtMethod);
+
+                ////internal static unsafe ref uint GetOrSetAt(this ref MainAVIHeader.__dwReserved_4 @this, int index)
```

Comment:
Could this be named `ItemRef` to match the BCL? https://github.com/dotnet/corefx/pull/25738

Reply:
Good idea. I was hurting for a good name.

Reply:
I just saw `GetAt` and realized that they can't both be called `ItemRef`.

Reply:
Right. I have to use unique names. I wonder which should be called `ItemRef`. In your PR, those are all immutable collections, but the name `ItemRef` doesn't *suggest* it is immutable.

Reply:
I'm inclined to go with `ItemRef` on the read/write ref. But what to rename `GetAt` to? `ItemReadRef`?

Reply:
`ItemRef`/`ReadOnlyItemRef` is a mouthful. Which of the two will be typed most often?

Reply:
The r/w will be used most often. If someone actually stores one of these structs in a `readonly` field, that's when they'll be forced to use the readonly method.

Reply:
I refreshed the page and your last three comments appeared. Sorry about that! I like the direction you went with it.